macros/src/main/scala-3/com/softwaremill/macwire/MacwireMacros.scala

```diff
@@ -2,16 +2,40 @@ package com.softwaremill.macwire
 
 import com.softwaremill.macwire.internals.*
 import scala.quoted.*
-import java.sql.Types
 
 object MacwireMacros {
   private val log = new Logger()
 
   def wireImpl[T: Type](using q: Quotes): Expr[T] = {
     import q.reflect.*
 
-    val constructorCrimper = new ConstructorCrimper[q.type, T](log)
-    val companionCrimper = new CompanionCrimper[q.type, T](log)
+    val dependencyResolver = DependencyResolver.throwErrorOnResolutionFailure[q.type, T](log)
+
+    wire[T](using q)(dependencyResolver)
+  }
+
+  //TODO build failure path 
+  def wireRecImpl[T: Type](using q: Quotes): Expr[T] = {
+    import q.reflect.*
+
+    // FIXME for some reason `TypeRepr.of[String].typeSymbol.owner` and `defn.JavaLangPackage` have different hash codes 
```

Comment:
maybe these are not case classes but ordinary class instances, and you simply get a different hash. Not sure if that's something we'll be able to fix

Reply:
I left this comment, because I had found a similar solution in [magnolia](https://github.com/softwaremill/magnolia/blob/7451e48b9a97808ef84722d6d2a0ad41db8e6e58/src/core/macro.scala#L152) so I though that maybe we could fix it.

Reply:
hm I wonder if the mangolia one works then ;)